### PR TITLE
Switch to SSA

### PR DESCRIPTION
**Pull request by inteon**
Switched both the `CertificateRequest` and `CertificateRequestPolicy` controllers to SSA.
This also required the tests to be updated (the fake client does not support mocking SSA).

### CODE REVIEW

**Review by irbekrm — APPROVED** [review]
Thanks @inteon I've read through the code- all makes sense to me and seems to be the same approach that we use elsewhere. This project does not have the same problem with moving to SSA as the other ones as we don't remove anything from CRPs/CRs.

I've played with the code a bit upgrading from an older version of approver-policy to this PR and checking that existing resources get correctly updated - all seems to work as expected.

I'm adding an approve with a hold as this PR hasn't been around for too long - in case anyone else wants to take a look, but happy for it to get merged.

Additionally, I think we need to update approver-policy docs to state that the oldest supported k8s version will be 1.22

/hold

**Comment by irbekrm** [comment]
Thanks for fixing the boilerplate check as well!

/lgtm
/unhold

File: pkg/internal/controllers/certificaterequests.go
Comment (wallrj) [inline pkg/internal/controllers/certificaterequests.go:153]: Typo in error message.

Should say "failed to generate CertificateRequest.Status patch.

**Pull request merged**
